api/datastore/sql/migrations/migrations.go

```diff
@@ -1,534 +1,136 @@
-		__7_add_route_cpusDownSql,
-		"7_add_route_cpus.down.sql",
-	)
-}
-
-func _7_add_route_cpusDownSql() (*asset, error) {
-	bytes, err := _7_add_route_cpusDownSqlBytes()
-	if err != nil {
-		return nil, err
+	// now that we're sorted in the appropriate direction,
+	// populate next and previous for each migration
+	for i, m := range migrations {
+		prev := int64(-1)
+		if i > 0 {
+			prev = migrations[i-1].Version
+			migrations[i-1].Next = m.Version
+		}
+		migrations[i].Previous = prev
 	}
 
-	info := bindataFileInfo{name: "7_add_route_cpus.down.sql", size: 37, mode: os.FileMode(420), modTime: time.Unix(1515624756, 0)}
-	a := &asset{bytes: bytes, info: info}
-	return a, nil
-}
-
-var __7_add_route_cpusUpSql = []byte("\x1f\x8b\x08\x00\x00\x00\x00\x00\x00\xff\x72\xf4\x09\x71\x0d\x52\x08\x71\x74\xf2\x71\x55\x28\xca\x2f\x2d\x49\x2d\x56\x70\x74\x71\x51\x48\x2e\x28\x2d\x56\xc8\xcc\x2b\xb1\xe6\x02\x04\x00\x00\xff\xff\xf1\x18\xf8\xa9\x21\x00\x00\x00")
-
-func _7_add_route_cpusUpSqlBytes() ([]byte, error) {
-	return bindataRead(
-		__7_add_route_cpusUpSql,
-		"7_add_route_cpus.up.sql",
-	)
+	return migrations
 }
 
-func _7_add_route_cpusUpSql() (*asset, error) {
-	bytes, err := _7_add_route_cpusUpSqlBytes()
-	if err != nil {
-		return nil, err
-	}
+func DownAll(driver string, db *sql.DB) error {
+	goose.SetDialect(driver)
+	migrations = sortAndConnectMigrations(migrations)
 
-	info := bindataFileInfo{name: "7_add_route_cpus.up.sql", size: 33, mode: os.FileMode(420), modTime: time.Unix(1515628068, 0)}
-	a := &asset{bytes: bytes, info: info}
-	return a, nil
-}
-
-// Asset loads and returns the asset for the given name.
-// It returns an error if the asset could not be found or
-// could not be loaded.
-func Asset(name string) ([]byte, error) {
-	cannonicalName := strings.Replace(name, "\\", "/", -1)
-	if f, ok := _bindata[cannonicalName]; ok {
-		a, err := f()
+	for {
+		currentVersion, err := goose.GetDBVersion(db)
 		if err != nil {
-			return nil, fmt.Errorf("Asset %s can't read by error: %v", name, err)
+			return err
 		}
-		return a.bytes, nil
-	}
-	return nil, fmt.Errorf("Asset %s not found", name)
-}
 
-// MustAsset is like Asset but panics when Asset would return an error.
-// It simplifies safe initialization of global variables.
-func MustAsset(name string) []byte {
-	a, err := Asset(name)
-	if err != nil {
-		panic("asset: Asset(" + name + "): " + err.Error())
-	}
-
-	return a
-}
-
-// AssetInfo loads and returns the asset info for the given name.
-// It returns an error if the asset could not be found or
-// could not be loaded.
-func AssetInfo(name string) (os.FileInfo, error) {
-	cannonicalName := strings.Replace(name, "\\", "/", -1)
-	if f, ok := _bindata[cannonicalName]; ok {
-		a, err := f()
+		current, err := migrations.Current(currentVersion)
 		if err != nil {
-			return nil, fmt.Errorf("AssetInfo %s can't read by error: %v", name, err)
+			fmt.Printf("goose: no migrations to run. current version: %d\n", currentVersion)
+			return nil
```

Comment:
should we check for a specific error here? should we return this error? unclear if `return nil` is in err

Reply:
fixex